crates/common_types/src/payments.rs

```diff
@@ -38,3 +40,35 @@ pub struct StripeSplitPaymentRequest {
     pub transfer_account_id: String,
 }
 impl_to_sql_from_sql_json!(StripeSplitPaymentRequest);
+
+#[derive(
+    Serialize, Deserialize, Debug, Clone, PartialEq, Eq, FromSqlRow, AsExpression, ToSchema,
+)]
+#[diesel(sql_type = Jsonb)]
+#[serde(deny_unknown_fields)]
+/// Hashmap to store mca_id's with product names
+pub struct AuthenticationConnectorAccountMap(
+    HashMap<enums::AuthenticationProduct, common_utils::id_type::MerchantConnectorAccountId>,
+);
+impl_to_sql_from_sql_json!(AuthenticationConnectorAccountMap);
+
+impl AuthenticationConnectorAccountMap {
+    /// get inner hashmap
+    pub fn inner(
+        &self,
+    ) -> &HashMap<enums::AuthenticationProduct, common_utils::id_type::MerchantConnectorAccountId>
+    {
+        &self.0
+    }
+    /// fn to get click to pay connector_account_id
+    pub fn get_click_to_pay_connector_account_id(
+        &self,
+    ) -> Result<common_utils::id_type::MerchantConnectorAccountId, errors::ValidationError> {
+        self.inner()
+            .get(&enums::AuthenticationProduct::ClickToPay)
+            .ok_or(errors::ValidationError::MissingRequiredField {
+                field_name: "authentication_product_id".to_string(),
```

Comment:
This error should say that, ClickToPay mca_id is missing

Reply:
but the missing required field in business profile is `authentication_product_id`